src/main/java/net/rptools/maptool/client/functions/MarkDownFunctions.java

```diff
@@ -47,46 +55,102 @@ public class MarkDownFunctions extends AbstractFunction {
           .setFrom(ParserEmulationProfile.GITHUB_DOC)
           .set(TablesExtension.COLUMN_SPANS, false)
           .set(TablesExtension.APPEND_MISSING_COLUMNS, true)
           .set(TablesExtension.DISCARD_EXTRA_COLUMNS, true)
-          .set(TablesExtension.HEADER_SEPARATOR_COLUMN_MATCH, true)
-          .set(com.vladsch.flexmark.parser.Parser.EXTENSIONS, extensions);
+          .set(TablesExtension.HEADER_SEPARATOR_COLUMN_MATCH, true);
     } else {
       options.setFrom(profile);
     }
 
+    // Cleanse optionalExtensions into an array to allow exact matches later on.
+    List<String> optionalExtensionsList =
+        new ArrayList<>(Arrays.asList(optionalExtensions.toUpperCase().trim().split("\\s*,\\s*")));
+
+    if (!optionalExtensionsList.isEmpty()) {
+      if (optionalExtensionsList.contains("*")) {
+        // Add all optional extensions.
+        extensions.add(AsideExtension.create());
+        extensions.add(AttributesExtension.create());
+        extensions.add(DefinitionExtension.create());
+        extensions.add(InsExtension.create());
+        extensions.add(StrikethroughSubscriptExtension.create());
+        extensions.add(SuperscriptExtension.create());
+        extensions.add(TablesExtension.create());
+        extensions.add(TaskListExtension.create());
+        extensions.add(TocExtension.create());
+      } else {
+        // Add user selected optional extensions
+        if (optionalExtensionsList.contains("ASIDE")) {
+          extensions.add(AsideExtension.create());
+        }
+        if (optionalExtensionsList.contains("ATTRIBUTES")) {
+          extensions.add(AttributesExtension.create());
+        }
+        if (optionalExtensionsList.contains("DEFINITION")) {
+          extensions.add(DefinitionExtension.create());
+        }
+        if (optionalExtensionsList.contains("INS")) {
+          extensions.add(InsExtension.create());
+        }
+        // Can only add the one Strikethrough and/or Subscript extension, so work out which one.
+        if (optionalExtensionsList.contains("STRIKETHROUGH")
+            && optionalExtensionsList.contains("SUBSCRIPT")) {
+          extensions.add(StrikethroughSubscriptExtension.create());
+        } else if (optionalExtensionsList.contains("STRIKETHROUGH")) {
+          extensions.add(StrikethroughExtension.create());
+        } else if (optionalExtensionsList.contains("SUBSCRIPT")) {
+          extensions.add(SubscriptExtension.create());
+        }
+        if (optionalExtensionsList.contains("SUPERSCRIPT")) {
+          extensions.add(SuperscriptExtension.create());
+        }
+        if (optionalExtensionsList.contains("TABLES")) {
+          extensions.add(TablesExtension.create());
+        }
+        if (optionalExtensionsList.contains("TASKLIST")) {
+          extensions.add(TaskListExtension.create());
+        }
+        if (optionalExtensionsList.contains("TOC")) {
+          extensions.add(TocExtension.create());
+        }
+      }
+    }
+
```

Comment:
This would be better handled with an enum so the code doesn't need to be added in two places if there is a new option.